### ISSUE TITLE
Wrong tokenization for distilbert-base-multilingual-cased

### ISSUE THREAD

**Issue by ricardorei** [huggingface/transformers#3594]
# 🐛 Bug

## Information

Model I am using (DistillBert):

The problem arises when using:
* my own modified scripts: (give details below)

The tasks I am working on is:
*  my own task or dataset: (give details below)

## To reproduce
with transformers 2.3.0:

```python
import torch
from transformers import DistilBertTokenizer
tokenizer = DistilBertTokenizer.from_pretrained("distilbert-base-multilingual-cased")
result = torch.tensor(tokenizer.encode("Hello, my dog is cute"))
print (result)

itos = tokenizer.ids_to_tokens
print (itos[61694])
print (itos[10133])

# The original token for 'Hello' exits but for some reason it's not used?
print (itos[31178])
```

Output:
```bash
[101, 61694, 10133, 117, 15127, 17835, 10124, 21610, 10112, 102]
'hell'
'##o'
'Hello'
```

## Expected behavior
```python
import torch
from transformers import DistilBertTokenizer
tokenizer = DistilBertTokenizer.from_pretrained("distilbert-base-multilingual-cased")
result = torch.tensor(tokenizer.encode("Hello, my dog is cute"))
print (result)
```

Output:
```bash
[101, 31178, 117, 15127, 17835, 10124, 21610, 10112, 102]
```

## Environment info

     
- `transformers` version: 2.3.0
- Python version: >3.6

**Comment by BramVanroy** [comment]
Can you show how you initialize `tokenizer`? Which vocab are you using?

**Comment by ricardorei** [comment]
> Can you show how you initialize `tokenizer`? Which vocab are you using?

sorry I forgot that... I updated the code in the issue already.

I was using `distilbert-base-multilingual-cased`

`tokenizer = DistilBertTokenizer.from_pretrained("distilbert-base-multilingual-cased")`

**Comment by LysandreJik** [comment]
This behaviour seems to have been solved in v2.7.0 as running your code yields the correct result in my environment.
